File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int l = 0;
        int r = 1;
        if (n == 1) answer = 1;
        else {
        
        while(true) {
            int ans = 0;
            if (l >= r) {
                break;
            }
            for (int i = l; i<=r; i++) {
                ans += i;
            }
            
            if (ans == n) {
                answer++;
                r++;
            } else {
                if (ans > n) {
                    l++;
                } else if (ans < n) {
                    r++;
                }
            }
        }
        answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    l, r, sum = 1, 1, 1
    
    while r <= n :
        if sum == n :
            answer+=1
        
        if sum >= n :
            sum -= l
            l += 1
        else :
            r += 1
            sum += r
    return answer